google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/UpdateBuilder.java

```diff
@@ -342,11 +297,14 @@ private T performSet(
    * @return The instance for chaining.
    */
   @Nonnull
-  public T update(
-      @Nonnull DocumentReference documentReference, @Nonnull Map<String, Object> fields) {
+  public T update(@Nonnull DocumentReference documentReference, @Nonnull Object fields) {
+    Object data = CustomClassMapper.convertToPlainJavaTypes(fields);
```

Comment:
As stated above, the top-level `fields` value should be a Map, but individual values can be custom POJOs.